src/pg_cron.c

```diff
@@ -265,6 +398,12 @@ PgCronWorkerMain(Datum arg)
 	/* Make pg_cron recognisable in pg_stat_activity */
 	pgstat_report_appname("pg_cron scheduler");
 
+	/*
+	 * Mark anything that was in progress before the database restarted as
+	 * failed.
+	 */
+	CleanAuditTable();
+
 	/* Determine how many tasks we can run concurrently */
 	if (MaxConnections < MaxRunningTasks)
```

Comment:
In the UseBackgroundWorkers case, we cannot have more than `max_worker_processes - 1` tasks running at a time (in the most optimistic scenario that pg_cron launcher is the only other background worker). It's probably lower in practice, but this at least prevents some issues.

I'm also wondering whether we should change the default value of MaxRunningTasks in the UseBackgroundWorkers case to some relatively low number.